server/src/main/java/org/elasticsearch/index/codec/CodecSupplier.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.codec;
+
+import org.apache.lucene.codecs.Codec;
+
+/**
+ * Abstracts codec lookup by name, to make CodecService extensible.
+ */
+@FunctionalInterface
+public interface CodecSupplier {
```

Comment:
How about CodecLookup or CodecProvider, but totally nit and no strong opinion

Reply:
I like CodecProvider!